src/map/clif.c

```diff
@@ -6833,6 +6833,8 @@ static void clif_item_refine_list(struct map_session_data *sd)
 	if (c > 0) {
 		sd->menuskill_id = WS_WEAPONREFINE;
 		sd->menuskill_val = skill_lv;
+	} else {	// If no items to refine
+		sd->state.prerefining = 0;
```

Comment:
i not sure but may be here need call ``clif_menuskill_clear``

Reply:
I tried adding this:
ShowDebug("menu skill id value is %d \n", sd->menuskill_id);
ShowDebug("menu skill val value is %d \n", sd->menuskill_val);

if c has no value, sd->menuskill_id and sd->menuskill_val is not set

result will be 0